### PR TITLE
added hw-3-2

### PR DESCRIPTION
**Pull request by kenu21**

### CODE REVIEW

File: src/main/java/core/basesyntax/Rectangle.java [hunk 1]
```diff
@@ -27,4 +27,16 @@ public Integer getLength() {
     public String getColor() {
         return color;
     }
+
+    public boolean equals(Rectangle ob) {
```
Comment (boroda4436) [inline src/main/java/core/basesyntax/Rectangle.java]: @kenu21 did you override this method from object?

File: src/main/java/core/basesyntax/Rectangle.java [hunk 2]
```diff
@@ -27,4 +27,16 @@ public Integer getLength() {
...
+        return (width.intValue() == ob.getWidth()) && (length.intValue() == ob.getLength());
+
+    }
+
+    public int hashCode() {
+        int result = 31;
```
Comment (boroda4436) [inline src/main/java/core/basesyntax/Rectangle.java:49]: @kenu21 Ok. This is fine. Did you know why here is using 31?

**Review by boroda4436 — CHANGES_REQUESTED** [review]

File: src/main/java/core/basesyntax/Rectangle.java (hunk 2 above)
Comment (kenu21) [reply src/main/java/core/basesyntax/Rectangle.java:49]: > @kenu21 Ok. This is fine. Did you know why here is using 31?
I watched video about hashCode to our homework. It explained there. This is optimal number.

File: src/main/java/core/basesyntax/Rectangle.java (hunk 1 above)
Comment (kenu21) [reply src/main/java/core/basesyntax/Rectangle.java]: > did you override this method from object?
From Object? Yes, I did.

Comment (boroda4436) [reply src/main/java/core/basesyntax/Rectangle.java]: No) You didn't)

**Review by boroda4436 — CHANGES_REQUESTED** [review]

File: src/main/java/core/basesyntax/Rectangle.java (hunk 1 above)
Comment (kenu21) [reply src/main/java/core/basesyntax/Rectangle.java]: > No) You didn't)
And now?)

File: src/main/java/core/basesyntax/Rectangle.java [hunk 3]
```diff
@@ -27,4 +27,21 @@ public Integer getLength() {
     public String getColor() {
         return color;
     }
+
+    @Override
+    public boolean equals(Object ob) {
+        Rectangle r = (Rectangle)ob;
+        return (width.intValue() == r.getWidth()) && (length.intValue() == r.getLength()
+                && getColor().length() == r.getColor().length());
+
```
Comment (boroda4436) [inline src/main/java/core/basesyntax/Rectangle.java]: @kenu21 this empty line between last row and '}' is not required

File: src/main/java/core/basesyntax/Rectangle.java [hunk 4]
```diff
@@ -27,4 +27,21 @@ public Integer getLength() {
...
+    }
+
+    @Override
+    public int hashCode() {
+        int result = 31;
+        result = result + (width != null ? width : 0);
```
Comment (boroda4436) [inline src/main/java/core/basesyntax/Rectangle.java]: @kenu21 lets use multiplying by 31 for each param

**Review by boroda4436 — CHANGES_REQUESTED** [review]

File: src/main/java/core/basesyntax/Rectangle.java [hunk 5]
```diff
@@ -27,4 +27,21 @@ public Integer getLength() {
     public String getColor() {
         return color;
     }
+
+    @Override
+    public boolean equals(Object ob) {
+        Rectangle r = (Rectangle)ob;
```
Comment (boroda4436) [inline src/main/java/core/basesyntax/Rectangle.java:39]: @kenu21 what if ob == null?
What if ob is not a rectangle?
What if you have received the same object?

**Pull request closed without merging**